BCODB submit form

HadleyKing opened the pull request:
- Added a submission form from Galaxy to BioCompute Object Database (BCODB). 
- Links to tutorial on how to populate, download or export a BCO

## How to test the changes?
(Select all options that apply)
- [ ] I've included appropriate [automated tests](https://docs.galaxyproject.org/en/latest/dev/writing_tests.html).
- [x] This is a refactoring of components with existing test coverage.
- [X] Instructions for manual testing are as follows:
  1. [tutorial and explanation](https://docs.biocomputeobject.org/galaxy_quick_start/) for Galaxy BCO submission

## License
- [x] I agree to license these contributions under [Galaxy's current license](https://github.com/galaxyproject/galaxy/blob/dev/LICENSE.txt).
- [x] I agree to allow the Galaxy committers to license these and all my past contributions to the core galaxy codebase under the [MIT license](https://opensource.org/licenses/MIT). If this condition is an issue, uncheck and just let us know why with an e-mail to galaxy-committers@lists.galaxyproject.org.

HadleyKing committed:
Merge pull request #59 from galaxyproject/dev

synch to main

HadleyKing committed:
BCODB submit form

Updated to add form for submission of BCO to BCODB
	modified:   client/src/components/WorkflowInvocationState/WorkflowInvocationExportOptions.vue

HadleyKing committed:
Finish BCODB submit form
Changes to be committed:
	modified:   client/src/components/WorkflowInvocationState/WorkflowInvocationExportOptions.vue

davelopez reviewed (commented):
Thank you @HadleyKing!
It looks good, only a couple of little comments on my side.

Since there is much more logic now, it would be nice to place all this in its own component `BioComputeObjectExportCard.vue` (or something similar) under a new directory `components/WorkflowInvocationState/Export`. This will help maintain multiple export options when they are available in `WorkflowInvocationExportOptions.vue`.

I also noticed that you added a nice tutorial with screenshots in https://w3id.org/biocompute/tutorials/galaxy_quick_start. Please note that this UI may change slightly as we add new *invocation export plugins* and consolidate them to be more homogeneous.

HadleyKing committed:
Create `components/WorkflowInvocationState/Export`

RE: review on https://github.com/galaxyproject/galaxy/pull/14374
- created `components/WorkflowInvocationState/Export`
- created 'components/WorkflowInvocationStatecomponents/WorkflowInvocationState/WorkflowInvocationExportOptions.vue'/Export/BioComputeObjectExportCard.vue`
- modified 'components/WorkflowInvocationState/WorkflowInvocationExportOptions.vue'

Changes to be committed:
	new file:   client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue
	new file:   client/src/components/WorkflowInvocationState/Export/index.js
	modified:   client/src/components/WorkflowInvocationState/WorkflowInvocationExportOptions.vue

davelopez commented on client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue:
```diff
@@ -0,0 +1,174 @@
+            bco: {
+                type: Object,
+                required: true,
+            },
```
This is not needed, see my next review comment.
```suggestion
```

davelopez commented on client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue:
```diff
@@ -0,0 +1,174 @@
+            .catch((e) => {
+                this.errors.push(e);
+            });
+    },
```
I think this request should be made directly in the `submitForm` method, otherwise, whenever you expand a random workflow invocation from the list the request will be made even if you don't want to export anything.

HadleyKing replied:
Resolved with [1bcd5fa](https://github.com/galaxyproject/galaxy/pull/14374/commits/1bcd5fac4a09b7e2b3a92e32cdb0403877fb0453)

davelopez commented on client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue:
```diff
@@ -0,0 +1,174 @@
+            } else {
+                return `${this.index + 1}`;
+            }
+        },
```
This seems to be unused
```suggestion
```

HadleyKing committed:
Update client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue

This is not needed

Co-authored-by: David López <46503462+davelopez@users.noreply.github.com>

HadleyKing committed:
Update client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue

This seems to be unused

Co-authored-by: David López <46503462+davelopez@users.noreply.github.com>

HadleyKing committed:
Fix for PR review

RE: https://github.com/galaxyproject/galaxy/pull/14374#discussion_r932224979
Changes to be committed:
	modified:   client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue

HadleyKing committed:
run prettier
Changes to be committed:
	modified:   client/src/components/WorkflowInvocationState/Export/BioComputeObjectExportCard.vue

davelopez reviewed (approved):
Thank you for addressing the comments!
It would be nice to move the authorization data to user preferences in a follow-up and we can unify the style of the cards when we have more export plugins in place.
